models/Task.php

```diff
@@ -1143,4 +1153,70 @@ private function getWorker()
     {
         return User::findOne(['id' => $this->getAccount(Task::ACCOUNT_WORKER)->user_id])->guid;
     }
+
+    /**
+     * revert all transactions where task space account is the target account
+     */
+    public function refund()
+    {
+        $incomeTransactions = Transaction::findAll(['to_account_id' => $this->getAccount(Task::ACCOUNT_SPACE)->id]);
+
+        foreach ($incomeTransactions as $transaction) {
+            $transaction->delete();
```

Comment:
hmm. normally I think deleting transactions should be impossible to not mess up with the bookkepping. 
not sure how important that is for us, but can't we do a transfer back?

Reply:
@bumi please see my answer to mortadha's comment about this same issue.

Also for the x-coin module , humhub team is not using a wallet for example to hold current balance and for which we can debit or credit , they are simply making transactions and calculating the balance by making the difference between income and outcome transactions.